### PR TITLE
Be task2

### PR DESCRIPTION
**Pull request by aliaksei-lyzo**
HomeTask 2 is completed.

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -21,15 +36,45 @@ const app = express();
...
 });
 
-app.use('/', blogRouter);
+const checkAuth = (req, res, next) => {
+  if (req.isAuthenticated()) {
+    return next();
+  } else {
+    res.redirect('/users/login')
+  }
+};
```
Comment (leskeg) [inline app/app.js]: I'd write like:
```javascript
const checkAuth = (req, res, next) => req.isAuthenticated
  ? next()
  : res.redirect('/users/login');
```
You don't have to re-write, it is just different code styling :)

File: app/config/passport.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const passport = require('passport');
+const LocalStrategy = require('passport-local').Strategy;
+const bcrypt = require('bcryptjs');
+
+const User = require('../models/user');
+
+passport.use(new LocalStrategy(
+  function(username, password, done) {
```
Comment (leskeg) [inline app/config/passport.js]: I would just use arrow funcitons

File: app/config/passport.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    User.findOne({ username: username }, function (err, user) {
+      if (err) { return done(err); }
```
Comment (leskeg) [inline app/config/passport.js]: I think curly brackets are not mandatory here.

File: app/config/passport.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      if (!user) {
+        return done(null, false, { message: 'Incorrect username.' });
+      }
+      bcrypt.compare(password, user.password, (err, isMatch) => {
```
Comment (leskeg) [inline app/config/passport.js]:
```javascript
(err, isMatch) => isMatch 
  ? done(null, user)
  : done(null, false { message: 'Incorrect password.' });

File: app/routes/blogs.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+});
+// I'm using object.assign, which helps to increase performance and reusability of endpoint,
+// by allowing us to send not the full object notation, but only the changed parts.
+blogRouter.put('/:id', (req, res, next) => {
+  const newBlogData = req.body;
+  Blog.findById(req.params.id)
+  .then(blog => {
+    blog = Object.assign(blog, newBlogData);
+    return blog;
+  })
```
Comment (leskeg) [inline app/routes/blogs.js]: I think you could simplify
```javascript
.then(blog => Object.assign(blog, newBlogData))
.then(blog => blog.save())
...

File: app/routes/blogs.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    author: req.body.author,
+    body: req.body.message
+  };
+  const newBlog = new Blog(blog);
+  newBlog.save()
+  .then(() => res.status(200).send('Blog saved successfully'))
+  .catch((err) => next(err));
+});
+// I'm using object.assign, which helps to increase performance and reusability of endpoint,
+// by allowing us to send not the full object notation, but only the changed parts.
```
Comment (leskeg) [inline app/routes/blogs.js]: Excellent idea

File: app/views/blogs/blogs.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+const deleteBlogButton = document.getElementById('delete-blog-button');
+
+const renderBlog = data => {
+  const blogItem = document.createElement('tr');
+  blogItem.innerHTML =
+    `<th scope="row">${data._id}</th>
+      <td>${data.title}</td>
+      <td>${data.author}</td>
+      <td>${data.body}</td>
+    `;
```
Comment (leskeg) [inline app/views/blogs/blogs.js]: Maybe it is possible to use pug to compile to html string, so you will have only pug. Take a look, don't need to change if it becomes very hard

File: app/app.js (hunk 1 above)
Comment (aliaksei-lyzo) [reply app/app.js]: Rewrote. I like this way better

File: app/config/passport.js (hunk 2 above)
Comment (aliaksei-lyzo) [reply app/config/passport.js]: Fixed

File: app/config/passport.js (hunk 3 above)
Comment (aliaksei-lyzo) [reply app/config/passport.js]: Refactored

File: app/config/passport.js (hunk 4 above)
Comment (aliaksei-lyzo) [reply app/config/passport.js]: Refactored

File: app/routes/blogs.js (hunk 5 above)
Comment (aliaksei-lyzo) [reply app/routes/blogs.js]: Me be blind...

File: app/routes/blogs.js (hunk 6 above)
Comment (aliaksei-lyzo) [reply app/routes/blogs.js]: Thanks...

File: app/views/blogs/blogs.js (hunk 7 above)
Comment (aliaksei-lyzo) [reply app/views/blogs/blogs.js]: Ok

**Pull request merged**